src/pools/pool.ts

```diff
@@ -1,10 +1,17 @@
+import type { IWorker } from './abstract-pool'
+import { AbstractPool } from './abstract-pool'
+
 /**
  * Contract definition for a poolifier pool.
  *
  * @template Data Type of data sent to the worker. This can only be serializable data.
  * @template Response Type of response of execution. This can only be serializable data.
  */
-export interface IPool<Data = unknown, Response = unknown> {
+export interface IPool<
+  Worker extends IWorker,
+  Data = unknown,
+  Response = unknown
+> extends AbstractPool<Worker, Data, Response> {
```

Comment:
Wait what?! Why do you want to let an interface extend from an abstract class?
The abstract class is the implementation of an interface that provides default / basic stuff

Reply:
It's exactly the same trick you've used with the IDynamicPool interface definition you've made to make the poolReference use inside the dynamic choice functions works. I do not really like but to code and test the initial callbacks registration, that's enough.  

Since the goal here is to make an API for callbacks that can be reused for all pool types, all worker type, a proper internal interface must be done to allow a transparent pool reuse inside the callbacks.